tests/components/cover/test_rfxtrx.py

```diff
@@ -0,0 +1,216 @@
+"""The tests for the Rfxtrx cover platform."""
+import unittest
+
+from homeassistant.bootstrap import _setup_component
+from homeassistant.components import rfxtrx as rfxtrx_core
+
+from tests.common import get_test_home_assistant
+
+
+class TestRollershutterRfxtrx(unittest.TestCase):
+    """Test the Rfxtrx roller shutter platform."""
+
+    def setUp(self):
+        """Setup things to be run when tests are started."""
+        self.hass = get_test_home_assistant(0)
+        self.hass.config.components = ['rfxtrx']
+
+    def tearDown(self):
+        """Stop everything that was started."""
+        rfxtrx_core.RECEIVED_EVT_SUBSCRIBERS = []
+        rfxtrx_core.RFX_DEVICES = {}
+        if rfxtrx_core.RFXOBJECT:
+            rfxtrx_core.RFXOBJECT.close_connection()
+        self.hass.stop()
+
+    def test_valid_config(self):
+        """Test configuration."""
+        self.assertTrue(_setup_component(self.hass, 'cover', {
+            'cover': {'platform': 'rfxtrx',
+                      'automatic_add': True,
+                      'devices':
+                          {'0b1100cd0213c7f210010f51': {
+                              'name': 'Test',
+                              rfxtrx_core.ATTR_FIREEVENT: True}
+                           }}}))
+
+    def test_invalid_config1(self):
+        """Test configuration."""
+        self.assertFalse(_setup_component(self.hass, 'cover', {
+            'cover': {'platform': 'rfxtrx',
+                      'automatic_add': True,
+                      'devices':
+                          {'2FF7f216': {
+                               'name': 'Test',
+                               'packetid': '0b1100cd0213c7f210010f51',
+                               'signal_repetitions': 3}
+                           }}}))
+
+    def test_invalid_config2(self):
+        """Test configuration."""
+        self.assertFalse(_setup_component(self.hass, 'cover', {
+            'cover': {'platform': 'rfxtrx',
+                      'automatic_add': True,
+                      'invalid_key': 'afda',
+                      'devices':
+                          {'213c7f216': {
+                              'name': 'Test',
+                              'packetid': '0b1100cd0213c7f210010f51',
+                              rfxtrx_core.ATTR_FIREEVENT: True}
+                           }}}))
+
+    def test_invalid_config3(self):
+        """Test configuration."""
+        self.assertFalse(_setup_component(self.hass, 'cover', {
+            'cover': {'platform': 'rfxtrx',
+                      'automatic_add': True,
+                      'devices':
+                          {'213c7f216': {
+                              'name': 'Test',
+                              'packetid': 'AA1100cd0213c7f210010f51',
+                              rfxtrx_core.ATTR_FIREEVENT: True}
+                           }}}))
+
+    def test_invalid_config4(self):
```

Comment:
These could all be under a single `def test_invalid_config(self):` or if you want them separate, more detailed name of what is invalid